### PR TITLE
MM-9478: set autofocus on default section in account settings modal

### PR DESCRIPTION
**Pull request by manland**
#### Summary

Set focus to the first Edit button of a tab of Account Settings
- General (Edit of Full Name)
- Security (Edit of Full Password)
- Notifications (Edit of Desktop notifications)
- Display (Edit of Theme)
- Sidebar (Edit of Group unread channels)
- Advanced (Edit of Send messages on CTRL+ENTER)

mattermost-autofocus-account-settings

#### Ticket Link

https://github.com/mattermost/mattermost-server/issues/10189

#### Checklist

- [X] Ran `make check-style` to check for style errors (required for all pull requests)
- [X] Ran `make test` to ensure unit and component tests passed
- [X] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: components/user_settings/modal/user_settings_modal.jsx [hunk 1]
```diff
@@ -76,7 +76,7 @@ class UserSettingsModal extends React.Component {
         this.state = {
             active_tab: 'general',
             active_section: '',
-            prev_active_section: '',
+            prev_active_section: 'dummySectionName',
```
Comment (lieut-data) [inline components/user_settings/modal/user_settings_modal.jsx]: Can you elaborate on the thinking behind `dummySectionName` vs. checking for just the empty string?

File: components/user_settings/display/user_settings_display.jsx [hunk 2]
```diff
@@ -682,6 +682,7 @@ export default class UserSettingsDisplay extends React.Component {
                         setRequireConfirm={this.props.setRequireConfirm}
                         setEnforceFocus={this.props.setEnforceFocus}
                         allowCustomThemes={this.props.allowCustomThemes}
+                        focused={this.props.prevActiveSection === this.prevSections.theme}
```
Comment (lieut-data) [inline components/user_settings/display/user_settings_display.jsx]: Can you elaborate on why special casing is required?

File: components/user_settings/modal/user_settings_modal.jsx (hunk 1 above)
Comment (manland) [reply components/user_settings/modal/user_settings_modal.jsx]: I don't know why but it's the default section title for each section https://github.com/mattermost/mattermost-webapp/blob/b0839434d1ef02c4268ed5d4d4e4f72c2b37eed5/components/user_settings/general/user_settings_general.jsx#L82

File: components/user_settings/display/user_settings_display.jsx (hunk 2 above)
Comment (manland) [reply components/user_settings/display/user_settings_display.jsx]: Because `ThemeSetting` is a sub-component instead of `SettingItemMin` in other cases.

**Comment by manland** [comment]
@lieut-data Hi, thanks for your review. 

It's my first ever react dev :tada: 

I'm not able to fix the problem about re-render with react. Can you give me advice ? I understand the problem but I have no idea how to fix it.

I have try to fix the initial issue (focus on default section) by reusing the `prevActiveSection` but maybe its a bad choice ?

**Comment by lieut-data** [comment]
@manland, it's definitely a more complicated problem, as it requires tracking the focussed state somewhere (`this.state`?) in order to preserve it. The current implementation isn't actually much better: I can't explain the exact behaviour, but it seems to focus one of two items when an re-render occurs. So at the very least, your changes aren't making it any /worse/, and definitely improving the common case! I'd probably leave this suggestion for a separate PR, if we tackle it at all.

Also, thanks for the extra detail: I'll take a second pass now :)

File: components/user_settings/modal/user_settings_modal.jsx (hunk 1 above)
Comment (lieut-data) [reply components/user_settings/modal/user_settings_modal.jsx]: Ah, that "makes sense". (To be honest, this whole block of code is a little confusing, but that's not your fault!)

Can you add that comment to make this explicit?
```suggestion
            prev_active_section: 'dummySectionName', // dummy value that should never match any section name
```

**Comment by manland** [comment]
@lieut-data thanks for the review!

**Comment by jwilander** [comment]
Thanks @manland !

**Comment by manland** [comment]
Fourth PR merged on my birthday :tada: :fire: :gift:

**Pull request merged**